### PR TITLE
UI: Update Ember/Ember CLI to 3.20

### PR DESCRIPTION
**Pull request by backspace**
This doesn’t include Ember Data, as we are still back on 3.12.

Most changes are deprecation updates, linting fixes, and dependencies. It can
be read commit-by-commit, though many of them are mechanical and skimmable.
For the new linting exclusions, I’ve added them to the Tech Debt list.

The [decrease in test count](https://github.com/hashicorp/nomad/pull/9641#issuecomment-745537453) is because linting is no longer included in `ember test`.

There’s a new deprecation warning in the logs that can be fixed by updating Ember
Power Select but when I tried that it caused it to render incorrectly, so I decided to
ignore it for now and address it separately:

```
DEPRECATION: The use of the private `{{-in-element}}` is deprecated, please refactor to the public `{{in-element}}`. ('ember-basic-dropdown/templates/components/basic-dropdown-content.hbs' @ L1:C0)  [deprecation id: glimmer.private-in-element]
```

### CODE REVIEW

**Review by DingoEatingFuzz — APPROVED** [review]
This looks nice and straightforward and all the tests pass ✨ 

There are just a couple extra long lines that were probably introduced by tooling.

File: ui/app/components/agent-monitor.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default class AgentMonitor extends Component {
   isStreaming = true;
   logger = null;
 
-  @computed('level', 'client.id', 'server.id')
+  @computed('client.id', 'level', 'server.{id,region}')
```
[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/components/agent-monitor.js:26]: This is an autofix situation where it added the missing `server.region` dependent key and also brace expansion, when it rewrites it doesn’t care about existing ordering, I guess. I can change it to stay in the same order if preferable!

File: ui/app/components/exec/task-group-parent.js [hunk 2]
```diff
@@ -48,10 +48,7 @@ export default class TaskGroupParent extends Component {
   @mapBy('activeTasks', 'taskGroup') activeTaskGroups;
 
   @computed(
-    'taskGroup.name',
-    'activeTaskStates.@each.name',
-    'activeTasks.@each.name',
-    'activeTaskGroups.@each.name'
+    'activeTaskGroups.@each.name', 'activeTaskStates.@each.name', 'activeTasks.@each.name', 'taskGroup.{name,tasks}'
```
[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/components/exec/task-group-parent.js:51]: eep… it seems as though Prettier didn’t run on this autofix change (and others, probably…) I’ll add another commit to fix these 😞 thanks for pointing it out

[the remark this answers is not in the document]
Comment (backspace) [reply ui/app/components/exec/task-group-parent.js]: Okay, I added 7d2c00e with line length fixes for a few files. I wish I knew why this happened.

I found 23 other files with Prettier autofixes, could be a separate PR at some point 🤔

**Pull request merged**